### ISSUE TITLE
Sometimes after publishing a product, the product is "not found"

### ISSUE THREAD

**Issue by spencern** [reactioncommerce/reaction#2249]
When creating a new product, sometimes it becomes "not found" after publishing it.
I don't have perfect reproduction steps for this one yet, but it's happened to me several times.
Here's a video:

http://recordit.co/uLT7rKSA0l

### Versions
```
Node: 7.8.0
NPM: 4.5.0
Meteor Node: 4.8.2
Meteor NPM: 4.5.0
Reaction CLI: 0.8.1
Reaction: 1.1.1
Reaction branch: development
Docker: 17.03.1-ce
```

**Comment by zenweasel** [comment]
@spencern this has been happening to me as well but I have not been able to reliably reproduce it. It will still appear in the grid but when you go to the PDP I get the 404 page

**Comment by aaronjudd** [comment]
I've seen this, so it can be reproduced... but not sure the steps..

**Comment by spencern** [comment]
Was able to reproduce this on the release-1.3 branch

**Comment by zenweasel** [comment]
Making this minor because it's bad when it happens but doesn't happen very often

**Comment by prinzdezibel** [comment]
On place this can happen is in the short time between publishing and redirection: https://github.com/reactioncommerce/reaction/blob/ac07d02a971797863d0f593fc4608dd4b0a2611c/imports/plugins/included/product-detail-simple/client/containers/publish.js#L52

I'm refactoring this as part of #2246, because the page reload is not good anyway. Not unlikely, that this problem gets resolved as part of it.

**Comment by prinzdezibel** [comment]
One example, where it happens reliable is when navigating to ProductGrid before acutally publishing a change to the permalink. 
![unpulished_notfound](https://user-images.githubusercontent.com/1733229/36217143-f9491fd8-11b0-11e8-91d9-765ab188e24b.gif)
See the gif as example.

This is addressed through the following PR (#Placeholder for PR).

There's also another PR on how the changed handles are published as part of ticket #2246, which involves a tracker not being executed when the product handle changes. This is also one situation where the behavior described in this ticket can occur. With this PR applied, I haven't seen this specific problem any longer: (#Placeholder for other PR)

**Comment by prinzdezibel** [comment]
Fixed completely with  #2246 and #3748
